### ISSUE TITLE
Unit tests throw timeout error after upgrading to 0.10.0

### ISSUE THREAD

**Issue by shiprabehera** [Mashape/kong#2442]
*Note: GitHub issues should be reserved only for bug reports. Please use the
[Kong mailing list](https://groups.google.com/forum/#!forum/konglayer) or
[Gitter](https://gitter.im/Mashape/kong) for user support, development
questions, feature requests, etc.*

### Summary

This is a combination of 2 issues.
After upgrading from kong version 0.9.9 to 0.10.0, the lua unit tests of my custom plugin started throwing timeout error. They were executing fine before the version upgrade. Apparently, the default timeout is set to 10s, so on doing this
`before_each(function()
    client = helpers.proxy_client(15000)
  end)`
the tests were executed successfully. Could this be specific to my local environment?
All the existing unit tests also throw timeout error.
Recently, I had to modify my source code to use **resty.http** module for http requests. After this change, again the unit tests started throwing timeout error even after setting it > 10s. 

### Additional Details & Logs

- Kong version (`$ kong version`) : 0.10.0
- Kong debug-level startup logs (`$ kong start --vv`)
- Kong error logs (`<KONG_PREFIX>/logs/error.log`)
- Kong configuration (registered APIs/Plugins & configuration file)
- Operating System : Kong is running on Vagrant. Host OS is Windows 7.

**Comment by patricksanders** [comment]
This may be a duplicate of #2352.

Does setting `KONG_NGINX_DAEMON=on` make your tests work?

**Comment by p0pr0ck5** [comment]
@shiprabehera any update based on the feedback above?

**Comment by shiprabehera** [comment]
@patricksanders @p0pr0ck5 
This does not work.  I am still getting timeouts. Sorry about the late update.

**Comment by p0pr0ck5** [comment]
It's a bit hard to say what's happening with looking at the test files, but I don't think a GitHub issue is the appropriate place, as this isn't related to kong itself, but your custom plugin tests (from what I read). There have been other reports of odd/unexpected behavior with Vagrant + Windows setups, so I wouldn't be surprised if such an environment was at least partially responsible. It may also be that your use of `resty.http` is possibly conflicting with the busted helpers, but again it's hard to say for sure without debugging. One other suggestion I might have would be to use Travis CI to run the tests as well, to exclude your environment as a cause for timeouts.

**Comment by p0pr0ck5** [comment]
Closing after no response. Feel free to re-open here if this seems like an issue with Kong, or continue the conversation on the mailing list. Thanks!

**Comment by sfhardman** [comment]
Leaving this here, as it's possibly the same issue and may save someone else a few hours...  

I brought up the kong-vagrant box and tried running the /kong-plugin/spec tests.  These failed with a timeout.  Tried again increasing the timeout to 60 secs and got an error message: 

```failed to receive reply from UDP server 8.8.8.8:53: timeout```

Turns out Google Public DNS is blocked by the firewall in my organisation's network.  The workaround was to comment out a line in /kong/spec/kong_tests.conf:

```# dns_resolver = 8.8.8.8```
